src/jit.rs

```diff
@@ -418,25 +430,96 @@ fn emit_store_imm32(jit: &mut JitMemory, size: OperandSize, dst: u8, offset: i32
     };
 }
 
+enum Value {
+    Register(u8),
+    RegisterPlusConstant(u8, i64),
+    Constant(i64)
+}
+
 #[inline]
-fn emit_register_saving_before_call(jit: &mut JitMemory) {
-    for reg in CALLER_SAVED_REGISTERS.iter() {
+fn emit_rust_call(jit: &mut JitMemory, function: *const u8, arguments: &[(usize, Value)]) {
+    let mut saved_registers = CALLER_SAVED_REGISTERS.to_vec();
+
+    // Pass arguments via stack
+    for argument in arguments {
+        if argument.0 < ARGUMENT_REGISTERS.len() {
+            continue;
+        }
```

Comment:
We never pass constants or register plus constant via the stack?

Reply:
Don't need to so far. If only registers are passed, which are in `CALLER_SAVED_REGISTERS`, then we can get away by only reordering them in saved_registers.

Reply:
@Lichtso The reason I asked is that `Argument.index` seemed gratuitous, could the slice index be used directly and the parameters are passed in the correct order, so what was `Argument.index` not is just the parameters index in the slice?

Reply:
`Argument.index` and its position in the slice may differ. E.g. for argument displacement they can be shifted. Also, we need to order them correctly and it is a lot easier to hard code the order than to infer it dynamically.